File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.swift
func solution(_ strings:[String], _ n:Int) -> [String] {
    return strings.sorted {
        let c1 = $0[$0.index($0.startIndex, offsetBy: n)]
        let c2 = $1[$1.index($1.startIndex, offsetBy: n)]
        if c1 == c2 {
            return $0 < $1
        } else {
            return c1 < c2
        }
    }
}

File: 프로그래머스/1/12915. 문자열 내 마음대로 정렬하기/문자열 내 마음대로 정렬하기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>

using namespace std;

vector<string> solution(vector<string> strings, int n) {
    vector<string> answer;
    sort(strings.begin(), strings.end(), [n](string a, string b){
        if (a[n] == b[n]) {
            return a < b;
        }
        return a[n] < b[n];
    });
    for(auto s: strings) {
        answer.push_back(s);
    }
    return answer;
}